third_party/terraform/resources/resource_storage_bucket.go

```diff
@@ -995,20 +1010,53 @@ func expandBucketWebsite(v interface{}) *storage.BucketWebsite {
 	if v := website["main_page_suffix"]; v != "" {
 		w.MainPageSuffix = v.(string)
 	}
-
 	return w
 }
 
+// remove this on next major release of the provider.
 func expandIamConfiguration(d *schema.ResourceData) *storage.BucketIamConfiguration {
-	return &storage.BucketIamConfiguration{
-		ForceSendFields: []string{"BucketPolicyOnly"},
-		BucketPolicyOnly: &storage.BucketIamConfigurationBucketPolicyOnly{
-			Enabled:         d.Get("bucket_policy_only").(bool),
-			ForceSendFields: []string{"Enabled"},
-		},
+	var ret *storage.BucketIamConfiguration
+
```

Comment:
I assume you need the d.HasChange because a true value in state for one will still appear in d.Get even if it's not set in config? If that's the case, we can still simplify with:
```
enabled := false
if d.HasChange("bucket_policy_only") {
  enabled = d.Get("bucket_policy_only").(bool)
} else if [etc]
...
}

return &storage.BucketIamConfiguration{
[etc, just use the value of enabled here]
```

Can you also add a comment explaining why we need to check HasChange for each one?

Reply:
done